magic/core/src/main/java/link/magic/android/core/relayer/WebViewWrapper.kt

```diff
@@ -142,6 +144,7 @@ class WebViewWrapper internal constructor(context: Context, private val urlBuild
                 val json = Gson().toJson(response.response)
                 messageHandlers[response.response.id]?.let { it(json) }
                 messageHandlers.remove(response.response.id)
+                resetContext()
```

Comment:
I'm just thinking out loud.
* Payload resolves/rejects https://github.com/magiclabs/magic-android/pull/22/files#diff-18675645bff7aef95282e1aa45f22a9a47d9a47ac6172b99f6dbb0ab99512717R147
* Payload hangs. The context will be dropped anyway
* I can't think of a third case

Reply:
Can we do `mMutableContext.baseContext = mImmutableContext`  to drop the reference of activity context

Reply:
Not sure I follow you here, doesn't  `MAGIC_HANDLE_RESPONSE` suggest that the payload has resolved? Or is the concern the specific context getting reset? In which case I can probably try to figure out a way to determine which `payload_id` correlates to which context before reseting it.

Reply:
> Can we do mMutableContext.baseContext = mImmutableContext to drop the reference of activity context

Yes, but we still need to handle the [reference in the provider](https://github.com/magiclabs/magic-android/pull/22/files#diff-3baedda570cc7d00931fbc9bfdf5011330f0840757dd2365e59cdf03aecebf9fR50-R54).